pkg/storaged/client.js

```diff
@@ -1770,8 +1770,13 @@ client.export_mount_point_mapping = () => {
         for (const c of b.Configuration) {
             if (c[0] == "fstab") {
                 const dir = client.strip_mount_point_prefix(utils.decode_filename(c[1].dir.v));
+                const device = utils.decode_filename(b.PreferredDevice);
+                const type = utils.decode_filename(c[1].type.v);
+
                 if (dir)
-                    mpm[dir] = utils.decode_filename(b.PreferredDevice);
+                    mpm[dir] = device;
+                if (type === "swap")
+                    mpm[type] = device;
```

Comment:
So what about `mpm[device] = { dir, type }`? I would be much happier with that.

Reply:
Yes! I like it.

Reply:
But you don't want to do this in this PR?